src/types/transactions.rs

```diff
@@ -468,6 +468,39 @@ pub enum Payload {
     },
 }
 
+impl Payload {
+    /// Resolve the TransactionType corresponding to the variant of the Payload.
```

Comment:
```suggestion
    /// Resolve the [TransactionType] corresponding to the variant of the Payload.
```

This will construct a link when generating documentation with `cargo doc`.